Files/Views/LayoutModes/GenericFileBrowser.xaml.cs

```diff
@@ -330,11 +330,20 @@ private void AllView_SelectionChanged(object sender, SelectionChangedEventArgs e
 
         private async void AllView_Sorting(object sender, DataGridColumnEventArgs e)
         {
+            bool hasNoSorting = e.Column.SortDirection == null;
+
             if (e.Column == SortedColumn)
                 App.CurrentInstance.FilesystemViewModel.IsSortedAscending = !App.CurrentInstance.FilesystemViewModel.IsSortedAscending;
```

Comment:
What about to add `e.Column.SortDirection = App.CurrentInstance.FilesystemViewModel.IsSortedAscending ? DataGridSortDirection.Ascending : DataGridSortDirection.Descending;` here instead of your code?

Reply:
If you switch the sorting column, the previous direction will be used. My code uses `DataGridSortDirection.Ascending` by default.

Reply:
Ok, then reset sorting by adding `e.Column.SortDirection = DataGridSortDirection.Ascending;` before if statement.

Reply:
```
e.Column.SortDirection = DataGridSortDirection.Ascending;
            if (e.Column == SortedColumn)
            {
                App.CurrentInstance.FilesystemViewModel.IsSortedAscending = !App.CurrentInstance.FilesystemViewModel.IsSortedAscending;
                e.Column.SortDirection = App.CurrentInstance.FilesystemViewModel.IsSortedAscending ? DataGridSortDirection.Ascending : DataGridSortDirection.Descending;
            }
            else if (e.Column != iconColumn)
            {
                SortedColumn = e.Column;
            }
```

Reply:
And for future, don't use negation in bool variable names, please.